mlir-tensorrt/executor/lib/Runtime/Backend/Lua/Modules/TensorRT/TensorRTModule.cpp

```diff
@@ -369,6 +449,27 @@ static Status enqueueV3Wrapper(AllocTracker &tracker,
 
   MTRT_DBGF("%s", "enqueueV3 successful and inputs are consumed");
 
+  if (hasResults) {
+    int64_t* desc = reinterpret_cast<int64_t*>(*outputDescriptors);
+
+    // Number of results are known in advance.
+    int64_t descIndex = 0;
+    int64_t nbResults = desc[descIndex++];
+    for (int64_t i = 0; i < nbResults; ++i) {
+      auto allocator = static_cast<CustomTensorRTOuputAllocator *>(
+          outputAllocatorTracker.get(i));
+      desc[descIndex++] = allocator->mOutputDims.size(); // rank
```

Comment:
Rank is always statically known. The compiler should generate executor IR to pre-populate the rank field, and this line should simply become a validation point -- read the pre-specified rank and compare with `mOutputDims.size()`. If they are different, then we should return an error, especially since the compiler generated an allocation that is only large enough to hold results with a specific rank.